arcgis-ios-sdk-samples/Augmented reality/Collect data in AR/CollectDataAR.swift

```diff
@@ -0,0 +1,574 @@
+            URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!)
+        scene.baseSurface?.elevationSources.append(elevationSource)
+
+        // Allow camera to go beneath the surface
+        scene.baseSurface?.navigationConstraint = .none
+
+        // Create a feature layer and add it to the scene
+        featureLayer = AGSFeatureLayer(featureTable: featureTable)
+        scene.operationalLayers.add(featureLayer!)
+        featureLayer?.sceneProperties?.surfacePlacement = .absolute
+
+        // Display the scene
+        arView.sceneView.scene = scene
+
+        // Create and add the graphics overlay for showing WIP new features
+        arView.sceneView.graphicsOverlays.add(graphicsOverlay)
+        graphicsOverlay.sceneProperties?.surfacePlacement = .absolute
+        graphicsOverlay.renderer = AGSSimpleRenderer(symbol: tappedPointSymbol)
+    }
+
+    // MARK: - View lifecycle management
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        // Start AR tracking; if we're local, only use the data source to get the initial position
+        arView.startTracking(realScaleModePicker.selectedSegmentIndex == 1 ? .initial : .continuous, completion: nil)
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        arView.stopTracking()
+    }
+
+    @IBAction func showCalibrationPopup(_ sender: UIBarButtonItem) {
+        if let controller = calibrationVC {
+            if realScaleModePicker.selectedSegmentIndex == 0 { // Roaming
+                isCalibrating = true
+            } else { // Local
+                isCalibrating.toggle()
+            }
+
+            if isCalibrating {
+                showPopup(controller, sourceButton: sender)
+            } else {
+                helpLabel.text = "Tap to record a feature"
+            }
+        }
+    }
+
+    @IBAction func addFeature(_ sender: UIBarButtonItem) {
+        if graphicsOverlay.graphics.count < 1 {
+            presentAlert(message: "You need to tap an object before you can save.")
+            return
+        }
+
+        if let coreVideoBuffer = arView.arSCNView.session.currentFrame?.capturedImage {
+            // Get image as useful object
+            // NOTE: everything here assumes photo is taken in portrait layout (not landscape)
+            var coreImage = CIImage(cvImageBuffer: coreVideoBuffer)
+            let transform = coreImage.orientationTransform(for: .right)
+            coreImage = coreImage.transformed(by: transform)
+            let ciContext = CIContext(options: nil)
+            let imageHeight = CVPixelBufferGetHeight(coreVideoBuffer)
+            let imageWidth = CVPixelBufferGetWidth(coreVideoBuffer)
+            let imageRef = ciContext.createCGImage(coreImage,
+                                                   from: CGRect(x: 0, y: 0, width: imageHeight, height: imageWidth))
```

Comment:
I believe so, as the image is rotated and there's a transform.  @nCastle1 can you confirm?

Reply:
That's what I needed to do, but I only tested in portrait. Looking at what's in the feature service, it may not work right when the device is in landscape.

Reply:
The code states it assumes everything is in Portrait.  If the user wants to adjust the code so it works in both, they can do that, but I think we're covered.